src/knoop/Node/src/insert.hpp

```diff
@@ -9,14 +9,14 @@ Node& insert( Str&& str, T&& t ){
 }
 
 template< typename T >
-Node& insert( list_iterator it, T&& t ){
+Node& insert( const list_iterator it, T&& t ){
```

Comment:
What if we don't have a `const` Node? Does this still work? You wrote a test to make sure the `const` iterator works. Can you write a test for a non-`const` iterator?

There are two main purposes for writing tests:
1. Testing the code (obviously)
2. Providing examples of how to use the code.

Reply:
Yes, this still works. In fact, the introduction of const has absolutely no effect on usage. Remember, iterators are like pointers. The questions of "is this iterator (pointer) const?" and "is the thing it iterates over (points to) const?" are absolutely independent of one another. Here, const simply makes explicit the fact that the variable called "it" doesn't change in the function - which it doesn't. It's a case of const correctness for the sake of const correctness, which I think most people would agree is a good thing. These functions are simple, and we can see that "it" doesn't change, but we might as well make it explicit. In a more complicated function, someone might send a parameter that they don't intend to change, and then make a mistake and try to change it. With const, the compiler detects the problem. If somebody really decides they want to change it, they just remove the const. So, in essence, const correctness is defensive programming.

My new test (of auto list() const) wasn't about const iterators, it was about iterators to const. (The Node to which list() is applied is const. Which should be - and now is - allowed, because one can extract a list from a const object; they just can't change the list thus extracted. Which is fine.)

Reply:
Thanks for the explanation. As I've said before, I avoid pointers because I don't like them. I don't like them because they are hard for me to understand.